tools/LLVMOptimizer.cpp

```diff
@@ -0,0 +1,42 @@
+#include "LLVMOptimizer.h"
+
+using namespace llvm;
+
+void optimize_module(llvm::StringRef optArgs, llvm::Module *M) {
+  llvm::LoopAnalysisManager LAM;
+  llvm::FunctionAnalysisManager FAM;
+  llvm::CGSCCAnalysisManager CGAM;
+  llvm::ModuleAnalysisManager MAM;
+  llvm::PassBuilder PB;
+
+  llvm::ModulePassManager MPM;
+
+  PB.registerModuleAnalyses(MAM);
+  PB.registerCGSCCAnalyses(CGAM);
+  PB.registerFunctionAnalyses(FAM);
+  PB.registerLoopAnalyses(LAM);
+  PB.crossRegisterProxies(LAM, FAM, CGAM, MAM);
+
+  if (optArgs == "O3") {
+    MPM = PB.buildPerModuleDefaultPipeline(OptimizationLevel::O3);
+  } else if (optArgs == "O2") {
+    MPM = PB.buildPerModuleDefaultPipeline(OptimizationLevel::O2);
+  } else if (optArgs == "O1") {
+    MPM = PB.buildPerModuleDefaultPipeline(OptimizationLevel::O1);
+  } else if (optArgs == "O0") {
+    MPM = PB.buildPerModuleDefaultPipeline(OptimizationLevel::O0);
+  } else if (optArgs == "Os") {
+    MPM = PB.buildPerModuleDefaultPipeline(OptimizationLevel::Os);
+  } else if (optArgs == "Oz") {
+    MPM = PB.buildPerModuleDefaultPipeline(OptimizationLevel::Oz);
+  } else {
+    if (auto E = PB.parsePassPipeline(MPM, optArgs)) {
+      llvm::errs() << E << "\n";
+    }
+  }
+  MPM.run(*M, MAM);
+}
+
+void optimize_function(llvm::StringRef optArgs, llvm::Function *func) {
+  optimize_module(optArgs, func->getParent());
```

Comment:
this is wrong. it will optimize the whole module. Plus it is never used in our codebase. Do you need it?

Reply:
You are right. This function is only necessary inside my mutate tool. I don't think Alive2 needs it. Maybe we could add it back in a better way once we need it.

Reply:
By the way, I know it is optimizing whole module, but I can't find a proper way to implement it. T_T So I just decide let it to optimize whole module instead.